xcube/webapi/ows/coverages/controllers.py

```diff
@@ -201,18 +269,26 @@ def _get_axis_properties(ds: xr.Dataset, dim: str) -> dict:
     )
 
 
-def _get_grid_limits_axis(ds: xr.Dataset, dim: str):
+def _get_grid_limits_axis(ds: xr.Dataset, dim: str) -> dict[str, Any]:
     return dict(
         type='IndexAxis', axisLabel=dim, lowerBound=0, upperBound=len(ds[dim])
     )
 
 
-def _get_units(ds: xr.Dataset, dim: str):
+def _get_units(ds: xr.Dataset, dim: str) -> str:
     coord = ds.coords[dim]
     if hasattr(coord, 'attrs') and 'units' in coord.attrs:
         return coord.attrs['units']
-    else:
-        return 'degrees'
+    if np.issubdtype(coord, np.datetime64):
+        return np.datetime_data(coord)[0]
+    if dim == 'time':
+        # Pure guesswork, but without an attribute or a datetime64 we can't
+        # do much better.
+        return 's'
+    # TODO: improve this guess -- can we match dataset dimensions to
+    #  projection axes?
+    # Fallback for non-time axes: use the unit of the first CRS axis
+    return pyproj.crs.CRS(get_crs_from_dataset(ds)).axis_info[0].unit_name
```

Comment:
Removed -- the method now returns "unknown" instead of trying to guess units from the CRS. We can discuss later whether there's any way to apply this approach more reliably, but better to stay safe and get the PR merged right now, I think.